query/options.go

```diff
@@ -3,11 +3,28 @@
 
 package query
 
-import "time"
+import (
+	"time"
+)
 
 type Options struct {
-	Start time.Time
-	End   time.Time
-	Step  time.Duration
-	Range time.Duration
```

Comment:
@fpetkovski Do we still need SelectRange here?

Reply:
I think we cannot have it here, because these are global options for the entire query. The range is really a parameter for a matrix selector and can be different for different selectors in a single query.

Reply:
Okay okay, so these options are not per operator, but for the whole query.
Then for expressions like sub query, we might need to use another start/end time to keep track of the iteration?

Reply:
Yes possibly. I am actually not that familiar with subqueries so I will also need to investigate more.